src/db_queries/admin_partners_students.ts

```diff
@@ -0,0 +1,175 @@
+import {Request, Response} from "express";
+import pg from "pg";
+import "moment-timezone";
+import {Conn} from "../db/conn";
+import {Helpers} from "../utils/helpers";
+import {PartnerStudentSearch} from "../models/partner_student_search.model";
+import {filterRowsBySearchParams} from "../models/partner_student.model.utils";
+import {PartnerStudent, StudentCertificationStatus} from "../models/partner_student.model";
+
+const conn = new Conn();
+const pool = conn.pool;
+const helpers = new Helpers();
+
+export class AdminPartnersStudents {
+  constructor() {
+    helpers.autoBind(this);
+  }
+
+  async getAllStudentsOfOnePartner(
+    request: Request,
+    response: Response
+  ): Promise<void> {
+    const {partner_id: partnerId} = request.params;
+    const searchParameters: PartnerStudentSearch = request.body;
+
+    const client = await pool.connect();
+    try {
+      await client.query("BEGIN");
+      const students = await this.getAllStudentsOfOnePartnerFromDB(
+        partnerId,
+        searchParameters,
+        client
+      );
+      response.status(200).json(students);
+      await client.query("COMMIT");
+    } catch (error) {
+      console.log("getAllStudentsOfOnePartner", error)
+      response.status(400).send(error);
+      await client.query("ROLLBACK");
+    } finally {
+      client.release();
+    }
+  }
+
+  private async getAllStudentsOfOnePartnerFromDB(
+    partnerId: string,
+    searchParameters: PartnerStudentSearch,
+    client: pg.PoolClient
+  ) {
+    const {
+      courseFromDate,
+      courseToDate,
+    } = searchParameters;
+    let whereFromToDateCondition = "";
+    let fromToDateCondition = "";
+
+    if (courseFromDate || courseToDate) {
+      if (courseFromDate && courseToDate) {
+        fromToDateCondition = `(
+          (uc.start_date_time between '${courseFromDate}' and '${courseToDate}')
+          or (uc.start_date_time + (ct.duration * INTERVAL '1 month') between '${courseFromDate}' and '${courseToDate}')
+       )
+      `;
+      } else if (courseFromDate && !courseToDate) {
+        // If only from date is provided, we will take all courses that end after that date
+        fromToDateCondition = `(uc.start_date_time + (ct.duration * INTERVAL '1 month') > '${courseFromDate}')`;
+      } else if (!courseFromDate && courseToDate) {
+        // If only to date is provided, we will take all courses that begin before that date
+        fromToDateCondition = `(uc.start_date_time < '${courseToDate}')`;
+      }
+      whereFromToDateCondition = ` where ${fromToDateCondition} `;
+    }
+    const allStudentsOfOnePartnerQuery = `
+      with
```

Comment:
The query looks good but it would be good to have the keywords in uppercase (we have recently done that on the mentors' side as well).

Reply:
done! check it out :)